scripts/paper_report.py

```
@@ -154,5 +268,92 @@ def _fmt_ratio_percent(value: float | None) -> str:
+def _fetch_api_json(
+    *,
+    api_base_url: str,
+    path: str,
+    api_token: str | None,
+) -> tuple[dict[str, Any], str | None]:
+    url = f"{api_base_url.rstrip('/')}{path}"
+    headers = {"Accept": "application/json"}
+    if api_token:
+        headers["Authorization"] = f"Bearer {api_token}"
+
+    request = Request(url, headers=headers)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)